src/components/AddSongMenu.tsx

```diff
@@ -0,0 +1,226 @@
+import Box from "@mui/material/Box";
+import Drawer from "@mui/material/Drawer";
+import Button from "@mui/material/Button";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemText from "@mui/material/ListItemText";
+import {
+  Avatar,
+  CircularProgress,
+  InputBase,
+  ListItemAvatar,
+  Paper,
+  Typography,
+} from "@mui/material";
+import MoreVertIcon from "@mui/icons-material/MoreVert";
+import { useContext, useState, FormEvent, ChangeEvent } from "react";
+import { AppContext } from "../App";
+import toast from "react-hot-toast";
+import CancelIcon from "@mui/icons-material/Cancel";
+import { handleExpiredToken } from "../helpers/handleExpiredToken";
+
+type AddMenuProps = {
+  currentSong: string;
+};
+
+interface PlayListWithTracksInterface
+  extends SpotifyApi.PlaylistObjectSimplified {
+  songsOnPlayList: SpotifyApi.PlaylistTrackObject[];
+}
+
+export default function AddSongMenu({ currentSong }: AddMenuProps) {
+  const { spotifyApi } = useContext(AppContext);
+  const [playListWithTracks, setPlayListWithTracks] = useState<
+    PlayListWithTracksInterface[]
+  >([]);
+  const [state, setState] = useState({
+    bottom: false,
+  });
+  const [playListName, setPlayListName] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const showFetchingDataMessage = <CircularProgress />;
+
+  async function fetchPlayList() {
```

Comment:
This is fetching multiple playlists so I'd recomment nameing it by the plurar ("`fetchPlayLists`").